controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/UpdateStreamTask.java

```diff
@@ -99,25 +99,27 @@ public UpdateStreamTask(final StreamMetadataTasks streamMetadataTasks,
 
         return Futures.toVoid(
                 streamMetadataStore.getEpochTransition(scope, stream, context, executor)
-                .thenCompose(etr -> streamMetadataStore.updateVersionedState(scope, stream, State.UPDATING, 
+                .thenCompose(etr -> streamMetadataStore.updateVersionedState(scope, stream, State.UPDATING,
                         state, context, executor)
                 .thenCompose(updated -> updateStreamForAutoStreamCut(scope, stream, configProperty, updated)
-                        .thenCompose(x -> notifyPolicyUpdate(context, scope, stream, configProperty.getStreamConfiguration(), 
+                        .thenCompose(x -> notifyPolicyUpdate(context, scope, stream, configProperty.getStreamConfiguration(),
                                 requestId))
-                        .thenCompose(x -> handleSegmentCounUpdates(scope, stream, configProperty, etr, context, executor, 
-                                requestId))
-                        .thenCompose(x -> streamMetadataStore.completeUpdateConfiguration(scope, stream, record, context, 
+                        .thenCompose(x -> handleSegmentCountUpdates(scope, stream, configProperty, etr, context, executor,
+                                                                    requestId))
+                        .thenCompose(x -> streamMetadataStore.removeTagsFromIndex(scope, stream, configProperty.getRemoveTags(), context, executor))
```

Comment:
remove tags from index and update tags to index can get some concurrency conflicts because multiple streams could be concurrently adding or removing tags. 
should we open an issue to think deeper about how we would want to solve it?

Reply:
Could you please create a github issue and put the issue id here. thanks

Reply:
https://github.com/pravega/pravega/issues/6096